neqo-transport/src/connection.rs

```diff
@@ -993,20 +1049,24 @@ impl Connection {
                     return Err(Error::VersionNegotiation);
                 }
                 (PacketType::Retry { odcid, token }, State::WaitInitial, Role::Client) => {
-                    if *odcid != self.dcid {
-                        qwarn!("received Retry, but not for us, dropping it");
+                    if self.retry_token.is_some() {
+                        qwarn!("received another Retry, dropping it");
                         return Ok(());
                     }
                     if token.is_empty() {
                         qwarn!("received Retry, but no token, dropping it");
                         return Ok(());
                     }
-                    if self.retry_token.is_some() {
-                        qwarn!("received another Retry, dropping it");
-                        return Ok(());
+                    {
+                        let path = self.paths.iter_mut().find(|p| p.remote_cid == *odcid);
+                        if path.is_none() {
+                            qwarn!("received Retry, but not for us, dropping it");
+                            return Ok(());
+                        }
+                        path.unwrap().remote_cid =
```

Comment:
avoiding `unwrap` with `if let Some()` will be fine.

Reply:
:+1: